### PR TITLE
Improve sign-up flow

### LINKED ISSUE #591: redesign "Log in" modal flow
One problem with the "Log in" modal (see screenshot below) is that it is used as a precursor for both for registrations and logins to already registered accounts, even though the text only talks about login rather than registration - so that's confusing to the user.

A second, consequent problem is that it is not a good place to ask for consent to process personal info, because we should only do that once per user at registration time (as per #245).

To elaborate: after clicking "Already been here? Log in", there is a checkbox asking for consent:

image

This kind of made sense when we had Facebook and Twitter logins enabled, because in that case a) login _was_ effectively registration, and b) there were no further screens on our side as part of the login process, so it made sense to gate login up front on obtaining consent.

But now we don't support Twitter/Facebook logins, only email.  And also we can't record consent in the db until we have an email address or some other form of identity, so it makes more sense to record consent at registration time, on the "Sign up with email" page.  Currently this page only asks for opt in to the newsletter, but it should really ask for consent to process personal info too.

### PR DESCRIPTION
**Pull request by stevebaxter**
This removes the "choice" dialog from the flow as we only support email now. We get data processing consent on the email sign-up form now.

We add "required" stars on all the forms that have required fields.

We also disable the timeout on the captcha as it can cause problems when the user clicks submit multiple times.

Closes #591

### CODE REVIEW

File: app/assets/stylesheets/home.scss [hunk 1]
```diff
@@ -74,7 +74,7 @@ select {
 }
 
 .container-extra-narrow {
-    max-width: 450px;
+    max-width: 500px;
```
Comment (stevebaxter) [inline app/assets/stylesheets/home.scss:77]: Make this a bit wider so the data processing consent checkbox fits

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
 
 class Users::RegistrationsController < Devise::RegistrationsController
+  include ApplicationHelper
+
   before_action :configure_sign_up_params, only: [:create]
-  invisible_captcha only: [:create]
+
+  # We disable timestamps because it can cause problems if the user clicks the submit button
+  # several times in a row
+  invisible_captcha only: [:create], timestamp_enabled: false
```
Comment (stevebaxter) [inline app/controllers/users/registrations_controller.rb:10]: I found this was a problem

File: app/views/devise/passwords/edit.html.erb [hunk 3]
```diff
@@ -11,12 +11,12 @@
         <%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
           <%= f.hidden_field :reset_password_token %>
 
-          <div class="form-group">
+          <div class="form-group required">
```
Comment (stevebaxter) [inline app/views/devise/passwords/edit.html.erb:14]: Adding `required` classes to these will add the red asterisk to their labels.

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -2,34 +2,46 @@
...
             <%= f.password_field :password_confirmation, autocomplete: "new-password", class: "form-control" %>
           </div>
 
           <div class="form-group form-check">
             <%= f.check_box :consent_news_email, class: "form-check-input" %>
-            <%= f.label  :consent_news_email, "Opt-in to Forward Democracy email updates", class: "mb-0 form-check-label" %>
+            <%= f.label  :consent_news_email, "Opt-in to Forward Democracy email updates", class: "form-check-label" %>
+          </div>
+
+          <div class="form-group form-check required">
```
Comment (stevebaxter) [inline app/views/devise/registrations/new.html.erb:35]: This is the new checkbox

File: app/views/home/_swap_form.html.haml [hunk 5]
```diff
@@ -40,7 +40,7 @@
     } else if (preferred_party_id == willing_party_id) {
       $(".js-equal-party-error-modal").modal();
     } else {
-      $("#js-login-modal").modal();
+       window.location.assign("#{sign_up_path}")
```
Comment (stevebaxter) [inline app/views/home/_swap_form.html.haml:43]: Just go straight to the sign up page now.

File: app/views/layouts/_login.html.haml [hunk 6]
```diff
@@ -1,59 +1,2 @@
 %span.user
-  = link_to "Already been here? Log in", "#js-login-modal", class: "small subdued logout", data: {toggle: "modal"}
-.modal#js-login-modal{ tabindex: "-1", role: "dialog",
```
Comment (stevebaxter) [inline app/views/layouts/_login.html.haml:3]: I've removed this modal completely as even if we turn FB and Twitter login back on we will need to come back to this flow.

File: spec/views/layouts/_login.html.haml_spec.rb [hunk 7]
```diff
@@ -1,82 +0,0 @@
-require "rails_helper"
```
Comment (stevebaxter) [inline spec/views/layouts/_login.html.haml_spec.rb:1]: This test is no longer needed because this modal is gone!

Comment (baob) [reply spec/views/layouts/_login.html.haml_spec.rb:1]: Seems we needed that test to keep our coverage up though :D

```
Coverage report generated for RSpec to /home/runner/work/swapmyvote/swapmyvote/coverage. 916 / 1316 LOC (69.6%) covered.
Line coverage (69.60%) is below the expected minimum coverage (70.00%).
Branch coverage ([56](https://github.com/swapmyvote/swapmyvote/actions/runs/9713695862/job/26811036363?pr=897#step:8:57).84%) is below the expected minimum coverage (58.00%).
SimpleCov failed with exit 2 due to a coverage related error
```

Comment (baob) [reply spec/views/layouts/_login.html.haml_spec.rb:1]: I was right, the relevant commit, for code coverage, says "set coverage thresholds so it only just passes".

@stevebaxter please adjust the thresholds a little bit in spec/spec_helper.rb.

**Pull request merged**